test/entities/test_arrays.py

```diff
@@ -0,0 +1,255 @@
+import numpy as np
+import pandas as pd
+import pandera as pa
+import pytest
+from numpy.testing import assert_allclose
+from pandas.testing import assert_frame_equal
+from pandera.typing import DataFrame
+
+from mechaphlowers.entities.arrays import SectionArray, SectionInputDataFrame
+
+
+@pytest.fixture
+def section_array_input_data() -> dict[str, list]:
+    return {
+        "name": ["support 1", "2", "three", "support 4"],
+        "suspension": [False, True, True, False],
+        "conductor_attachment_altitude": [2.2, 5, -0.12, 0],
+        "crossarm_length": [10, 12.1, 10, 10.1],
+        "line_angle": [0, 360, 90.1, -90.2],
+        "insulator_length": [0, 4, 3.2, 0],
+        "span_length": [1, 500.2, 500.05, np.nan],
+    }
+
+
+@pytest.fixture
+def section_array(section_array_input_data: dict[str, list]) -> SectionArray:
+    df = DataFrame[SectionInputDataFrame](section_array_input_data)
+    return SectionArray(data=df, sagging_parameter=2_000, sagging_temperature=15)
+
+
+def test_create_section_array(section_array_input_data: dict) -> None:
+    input_df: DataFrame[SectionInputDataFrame] = DataFrame(section_array_input_data)
+    section = SectionArray(input_df, sagging_parameter=2_000, sagging_temperature=15)
+
+    assert_frame_equal(input_df, section.data, check_dtype=False, rtol=1e-07)
+
+
+def test_create_section_array__span_length_for_last_support(
+    section_array_input_data: dict,
+) -> None:
+    section_array_input_data["span_length"][-1] = 300
+    input_df: DataFrame[SectionInputDataFrame] = DataFrame(section_array_input_data)
+
+    with pytest.raises(pa.errors.SchemaErrors):
+        SectionArray(input_df, sagging_parameter=2_000, sagging_temperature=15)
+
+
+def test_create_section_array__span_length_for_last_support_is_zero(
```

Comment:
@adriengoeller I'm not sure about the behavior we want in this case - that is, when a value is given in the last row of the input dataframe for `SectionArray`, in the column "span_length", and this value is zero. 

For now this raises an error, as a quite strict validation ensures that the last "span_length" is NaN, so I added this test so we know this is the current behavior. But it is debatable. The validation could be less strict and accept 0 as the last span_length. 

What do you think?

Reply:
Done